### PR TITLE
EW-315 Updating authentication routes

### PR DESCRIPTION
**Pull request by psachmann**
# Description
Updating authentication routes to use the new v3 endpoints for login.

## Links to Tickets or other pull requests
- <https://ticketsystem.dbildungscloud.de/browse/EW-315>
- <https://github.com/hpi-schul-cloud/schulcloud-server/pull/3744>

## Changes
See description and Jira ticket.

## Datasecurity
Does not apply.

## Deployment
Does not apply.

## New Repos, NPM pakages or vendor scripts
Does not apply.

## Screenshots of UI changes
Does not apply.

## Approval for review

- [x] All points were discussed with the ticket creator, support-team or product owner. The code upholds all quality guidelines from the PR-template.

> Notice: Please remove the WIP label if the PR is ready to review, otherwise nobody will review it.

### CODE REVIEW

File: helpers/authentication.js [hunk 1]
```diff
@@ -10,18 +9,7 @@ const isJWT = (req) => {
 };
 
 const isAuthenticated = (req) => {
-    if(!isJWT(req)) {
-        return Promise.resolve(false);
-    }
-
-    return api(req).post('/authentication', {json: {
```
Comment (HeftyTea) [inline helpers/authentication.js:17]: This allowed re-authentification in case a cookie was missing, right? Why was this removed

**Review by HeftyTea — CHANGES_REQUESTED** [review]

**Review by HeftyTea — CHANGES_REQUESTED** [review]

File: controllers/login.js [hunk 2]
```diff
@@ -14,8 +14,8 @@ router.post("/login/", function (req, res, next) {
   const password = req.body.password; // TODO: sanitize
 
   const login = (data) => {
-    return api(req)
-      .post("/authentication", { json: data })
+    return api(req, { version: 'v3' })
```
[the remark this answers is not in the document]
Comment (psachmann) [reply controllers/login.js:17]: Currently there is no implemented way to handle flags in the superhero dashboard. So we decided to hardcode these routes like in the previous refactoring's.

File: helpers/authentication.js (hunk 1 above)
Comment (psachmann) [reply helpers/authentication.js]: I misunderstood the code and reverted these changes.

**Review by HeftyTea — APPROVED** [review]
